docs/release_notes.rst

```diff
@@ -6,6 +6,14 @@ Unreleased
 
 This is a major release and contains breaking changes. Please read the notes below carefully.
 
+* :py:class:`~pynamodb.attributes.BinaryAttribute` and :py:class:`~pynamodb.attributes.BinarySetAttribute` have undergone breaking changes:
+
+  * The attributes' internal encoding has changed. To prevent this change going unnoticed, :code:`legacy_encoding` have been made required: see :doc:`upgrading_binary` for details.
+    If your codebase uses :py:class:`~pynamodb.attributes.BinaryAttribute` or :py:class:`~pynamodb.attributes.BinarySetAttribute`,
+    go over the attribute declarations and mark them accordingly.
+  * When using binary attributes, the return value of :py:func:`~pynamodb.model.Model.serialize` will no longer be JSON-serializable
+    since it will contain :code:`bytes` objects.
```

Comment:
```suggestion
  * When using binary attributes, the return value of :py:func:`~pynamodb.models.Model.serialize` will no longer be JSON-serializable
    since it will contain :code:`bytes` objects. Note that both `:py:func:`~pynamodb.models.Model.to_dict` and `:py:func:`~pynamodb.model.Model.to_json` are also affected.
```

Should we also include a suggestion to remediate?

Reply:
I'd like to implement those remediations first, won't release 6.0 without it.